### ISSUE TITLE
Training process hangs.

### ISSUE THREAD

**Issue by uyolo1314** [ultralytics/ultralytics#2300]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

![image](https://user-images.githubusercontent.com/101853326/235040028-1600292d-4efa-4c46-a576-0f73a9da27e2.png)
 When using default dataloader, it hangs. (I have tried many times with same seed, it hangs at same position)

But when using v5loader, it's ok!!!

### Environment

- YOLO: Ultralytics YOLOv8.0.87 🚀 Python-3.7.5 torch-1.10.0+cu111 CUDA:0 (GeForce RTX 3090, 24268MiB)
- OS: Ubuntu 18.04

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@uyolo1314 could you provide the command for both cases to reproduce?

**Comment by uyolo1314** [comment]
I just replace `valoader=False` with `valoader=True`. The dataset is not public, so it's hard to reproduce.

**Comment by uyolo1314** [comment]
I just replace `valoader=False` with `valoader=True`. The dataset is not public, so it's hard to reproduce.

**Comment by glenn-jocher** [comment]
@uyolo1314 thank you for reporting this issue. We apologize for the inconvenience this may have caused. 

To help us better understand the cause of this issue, could you please provide some additional information? Specifically, could you provide us with the command used to reproduce the issue with the default dataloader?

Should you have any other questions, please don't hesitate to ask - our team is more than happy to help!
